pmd-core/src/main/java/net/sourceforge/pmd/PMD.java

```diff
@@ -203,6 +203,13 @@ public static int doPMD(PMDConfiguration configuration) {
             return 0;
         }
 
+        // Log warning only once, if not explicitly disabled
+        if (!configuration.isIgnoreIncrementalAnalysis() && LOG.isLoggable(Level.WARNING)) {
```

Comment:
this warning shouldn't be produced if we configured a cache either

Reply:
Good catch!